pkg/xds/server/dataplane_metadata_tracker.go

```diff
@@ -0,0 +1,74 @@
+package server
+
+import (
+	"context"
+	"github.com/Kong/kuma/pkg/core/xds"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	go_cp_server "github.com/envoyproxy/go-control-plane/pkg/server"
+	"github.com/pkg/errors"
+	"sync"
+)
+
+type DataplaneMetadataTracker struct {
+	mutex            sync.RWMutex
+	metadataForProxy map[xds.ProxyId]*xds.DataplaneMetadata
```

Comment:
If we assume that `DataplaneMetadata` is unique per gRPC stream then it means that there might be multiple `DataplaneMetadata`s per `Dataplane` so how do we then choose which one to use for given `Dataplane`?

Reply:
and if we assume that `DataplaneMetadata` is unique per gRPC stream then we need to have to maps anyways, so we can associate Dataplane to stream.

Reply:
I think, the following changes could be a solution:
* change definition of `NewDataplaneWatchdogFunc`  to accept both `streamID` and `dataplaneId` as arguments
*  change `DataplaneMetadataTracker` to have `metadata map[int64]*xds.DataplaneMetadata` instead of 2 separate maps
* change definition of `DataplaneMetadataTracker.Metadata(..)` to accept `streamID` instead of `ProxyId`
* change `DefaultDataplaneSyncTracker(..)` to use `streamID` when calling `metadataTracker.Metadata(..)`

Reply:
Ok. Refactored to use stream id